### DISCUSSION TITLE
Refactor ApplicationService

### DISCUSSION THREAD

**Discussion by GOODBOY008** [apache/streampark#2980] [Ideas]
>###  `ApplicationService` is violating the Interface segregation principle.
>### Try to refactor this interface by breaking it down into smaller according to different principle.

 ### 1. Split `ApplicationService` into three interface by principle. 

- _`ApplicationService` :  manage application info._
- _`ApplicationUtilsService` : for validate and query special info._
- _`ApplicationOpService` : operate the jobs on cluster corresponding to application._

![image](https://github.com/apache/incubator-streampark/assets/13617900/983cfaa7-6fb7-47d9-b314-8ccb115ed51f)

### Appreciate any comments or feedback you may have on this discussion.

**Comment by wolfboys** [comment]
How about split ApplicationService into three interface : ApplicationManageService, ApplicationActionService, ApplicationInfoService?<img width="1496" alt="image" src="https://github.com/apache/incubator-streampark/assets/13284744/6c716400-25e1-4295-8d6f-32e28c1e69f4">
